plex/player.py

```diff
@@ -174,10 +176,48 @@ def _startPlayback(self):
             )
             self._duration = toMilliseconds(self.getTotalTime())
 
+            # register settings
+            settings = self._mediaProvider.prepareSettings()
+            if not settings:
+                log('failed to load settings for {} ({}) playing from {}' \
+                    .format(self._item.title, self._file, mediaProvider2str(self._mediaProvider)), xbmc.LOGWARNING)
+                self._reset()
+                return
+
+            # load external subtitles
+            if settings.getBool(SETTINGS_PROVIDER_PLAYBACK_ENABLE_EXTERNAL_SUBTITLES):
+                self._addExternalSubtitles(plexServer.PlexServer())
+
         else:
             self._reset()
 
 
+    def _addExternalSubtitles(self, plexServer):
+        '''Add external subtitles to the player'''
+        if not self._item:
+            return
+
+        external_subtitles = self._item.subtitleStreams()
+        if external_subtitles:
+            for subtitle in external_subtitles:
+                self.addSubtitle(
+                    plexServer.url(subtitle.key, includeToken=True),
+                    str(subtitle.title),
+                    str(subtitle.language),
+                    subtitle.selected
```

Comment:
IIRC correctly from reading the docs a while ago there's also a `forced` flag. Maybe we should consider that one as well (although it might not be possible on external subtitles).

Reply:
It has a forced flag but the kodi API doesn't allow to set it. IIRC kodi sets the Forced flag depending on the name of the subtitle. Plex has its own internal parser though so, for this to work another parameter need to be added to `xbmc.Player().addSubtitles`. I'd suggest a TODO for now.

Reply:
Added a TODO

Reply:
Thanks I have added this method to the Python API so I'll see about extending it with an optional `forced` parameter.